src/parse.yy

```diff
@@ -1083,6 +1094,22 @@ declaration_statement
 declaration
     : declaration_specifiers ';'
       {
+          DeclSpecs *ds = $1;
+          if (ds && ds->attributeList && ds->attributeList->HasAttribute("aligned")) {
+
+              std::string name;
+              const Type *t = ds->baseType;
+              if (CastType<StructType>(t)) {
+                  name = CastType<StructType>(t)->GetStructName();
+              } else if (CastType<UndefinedStructType>(t)) {
+                  name = CastType<UndefinedStructType>(t)->GetStructName();
+              }
+
+              if (!name.empty()) {
+                  Warning(@1, "attribute ignored in declaration of 'struct %s'. It must follow the 'struct' keyword.",
```

Comment:
Could you add a test case for this warning?

Reply:
It is covered in `tests/lit-tests/aligned-attr-warn.ispc`.